R/estimate_KM.R

```diff
@@ -65,101 +65,68 @@
 #' visR::estimate_KM(data = veteran_adam, strata = "trt")
 
 estimate_KM <- function(
-   data = NULL
-  ,strata = NULL
-  ,CNSR = "CNSR"
-  ,AVAL = "AVAL"
-  ,...
+    data = NULL
+    ,strata = NULL
+    ,CNSR = "CNSR"
+    ,AVAL = "AVAL"
+    ,...
 ){
 
-# Capture input to validate user input for data argument -----------------
-  dots <- rlang::dots_list(...)
-
- ## Get actual data name as symbol
- ### Magrittre pipe returns "." which inactivates recalls to survfit in downstream functions
- ### map passes .x as Call$data
- ### df: catch expressions that represent base R subsets
-  Call <- as.list(match.call())
-  dfExpr <- Call[["data"]]
-
- ## Validate `data` and capture data name
+  # Capture input to validate user input for data argument ---------------------
 
-  if (is.null(data)) stop(paste0("Data can't be NULL."))
-
-  if (base::length(base::deparse(Call[["data"]])) == 1 && base::deparse(Call[["data"]]) %in% c(".", ".x")){
-    df <- the_lhs()
-    Call[["data"]] <- as.symbol(df)
-  } else {
-     df <- as.character(sub("\\[.*$", "", deparse(dfExpr))[1])
-  }
+  dots <- rlang::dots_list(...)
 
-  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) stop(paste0("Data can be of `class` dataframe or tibble or data.table."))
+  # Validate argument inputs ---------------------------------------------------
 
-  data <- as.data.frame(data)
+  if (is.null(data))
+    stop(paste0("Data can't be NULL."))
 
-# Validate columns --------------------------------------------------------
+  if (!is.data.frame(data))
+    stop("Data does not have class `data.frame`.")
 
   reqcols <- c(strata, CNSR, AVAL)
-
   if (! all(reqcols %in% colnames(data))){
     stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols, colnames(data)), collapse = " "), "."))
   }
 
-  if (! is.numeric(data[[AVAL]])){
+  if (!is.numeric(data[[AVAL]])){
     stop("Analysis variable (AVAL) is not numeric.")
   }
 
-  if (! is.numeric(data[[CNSR]])){
+  if (!is.numeric(data[[CNSR]])){
     stop("Censor variable (CNSR) is not numeric.")
   }
 
-# Ensure the presence of at least one strata -----------------------------
-
-  if (is.null(strata)){
-    main <- "1"
-  } else {
-    main <- paste(strata, collapse = " + ")
-  }
-
-# Remove NA from the analysis --------------------------------------------
+  # Remove NA from the analysis ------------------------------------------------
 
-  data <- as.data.frame(data) %>%
-    tidyr::drop_na(AVAL, CNSR)
+  data <-
+    as.data.frame(data) %>%
+    tidyr::drop_na(any_of(c(AVAL, CNSR)))
 
   if (!is.null(strata)){
     data <- data %>%
       tidyr::drop_na(any_of({{strata}}))
```

Comment:
Remove the `{{ }}` here. The `strata` object can only be a character vector and you use the curly-curly when working the captured quos

Reply:
I removed the curly brackets